### ISSUE TITLE
"Tests not found"

### ISSUE THREAD

**Issue by enkeyz** [JohnnyHendriks/TestAdapter_Catch2#35]
So I got this message after I run the tests.

`[2019. 12. 06. 6:46:01.428 du.] ---------- Discovery started ----------
Test Adapter for Google Test: Test discovery starting...
Test discovery completed, overall duration: 00:00:00.0418552
Started Catch2Adapter test discovery...
Finished Catch2Adapter test discovery.
No test is available in C:\Users\enkeyz\source\repos\chapter10\x64\Debug\Listing10-1.exe. Make sure that test discoverer & executors are registered and platform & framework version settings are appropriate and try again.
[2019. 12. 06. 6:46:02.132 du.] ========== Discovery finished: 0 tests found (0:00:00,6366892) ==========
No tests found to run.
No tests found to run.`

**Comment by ScottHutchinson** [comment]
Note "Test Adapter for **Google Test**"

Try disabling Google Test and other test adapters than Catch2.

**Comment by enkeyz** [comment]
Same message. Disabled both Google and Boost Test adapters.

**Comment by ScottHutchinson** [comment]
Yeah, I've never been able to get this adapter to work with our projects in VS 2017. There are a few open issues regarding test discovery, which might have information that will help you.

**Comment by JohnnyHendriks** [comment]
To diagnose a little further what is going on you can do the following. Set logging to debug (_i.e._, add the following setting to the runsettings-file
```
<Logging>debug</Logging>
```
Note, with VS2019 you may need to switch configurations (_e.g._, from Debug to Release) and build the test between discovery actions otherwise VS2019 may think it does not need to discover as it may assume it has the tests in the cache.

Also, you could run the executable on the commandline with the following parameters
```
Listing10-1.exe --verbosity high --list-tests *
```
This is the default command line used by the test adapter when discovering tests.

**Comment by MPJansen** [comment]
Having a similar problem, the reference project detects the testcases but will not run them. Might there be something similar going on to [this](https://github.com/xkbeyer/CatchTestAdapter/issues/21)?

Thanks for your effort though!

**Comment by JohnnyHendriks** [comment]
Just released a new version (1.6.0) and cleaning up issues now. I do not know if there is anything more I can do with this, so closing the issue.

But just to add a little extra info (not related to the original issue). I've noticed that when switching configuration, and trying to run a test without re-building first, it will not run the test. It probably gets confused between the executables for the different configurations.
